main.cpp

```diff
@@ -245,39 +268,125 @@ INT
 */
 struct IntType
 {
-    int add(int lhs, int rhs);
-    int subtract(int lhs, int rhs);
-    int multiply(int lhs, int rhs);
-    int divide(int lhs, int rhs);
+    IntType(int intValue)
+    {
+        value = new int(intValue);
+    }
```

Comment:
If you have to use a constructor that takes parameters, and those parameters initialize member variables, you should always prefer the member initialize list compared to initializing the variables inside the constructor body.
The reason is because the variables need to be instantiated/initialized when the class is created, and then when you assign to them in the ctor body, you are potentially re-initializing them.  that is a waste of resources.
- [ ] revise, use the member initializer list.